feat(shared-data): Add module calibration adapter definitions.

vegano1 opened the pull request:
# Overview

This will add the labware definitions for the module calibration adapters that will be used to automatically calibrate the module offsets on the flex. These values are taken from the DVT Robot Extentns file [here](https://docs.google.com/spreadsheets/d/1yGRN-5QOa45p-jLsi4tL_PbHmKSK-QsXqHws6oBK7do/edit#gid=725031028) 


# Test Plan

- [x] Make sure we can load these using the standard load labware protocol engine command


# Changelog

- adds the labware definitions for the module calibration adapters

# Review requests

- Check that the offsets and geometries are accurate

# Risk assessment

low

vegano1 committed:
feat(shared-data): Add module calibration adapter definitions.

vegano1 committed:
fix typo

vegano1 committed:
set displayCategory to aluminumBlock
fix temperature module naming

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_heatershaker_module/1.json:
```diff
@@ -0,0 +1,53 @@
+    "brandId": []
+  },
+  "metadata": {
+    "displayName": "Opentrons Calibration Adapter - Heatershaker Module",
```
```suggestion
    "displayName": "Opentrons Calibration Adapter - Heater-Shaker Module",
```

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_heatershaker_module/1.json:
```diff
@@ -0,0 +1,53 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 0.25,
```
I don't think this is right - this should be the distance from the top plane of the "well" (the open part of the square) to its bottom.

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_magnetic_module/1.json:
```diff
@@ -0,0 +1,53 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 0.25,
```
see also heater shaker, don't think this is right

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_temperature_module/1.json:
```diff
@@ -0,0 +1,53 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 0.25,
```
don't think this is right

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_thermocycler_module/1.json:
```diff
@@ -0,0 +1,53 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 0.25,
```
seems wrong

sfoster1 reviewed (approved):
We should get some real depth numbers for these squares and in general I'm a little dubious of the exact numbers but if you need to get unblocked we can merge it

vegano1 committed:
add the actual dimension of the adapter + calibration square depth

vegano1 committed:
pretty lint

sfoster1 reviewed (changes requested):
Oh you know what we should do while we're in here - let's add a second well that is the z offset calibration point! This can be a circular well, maybe diameter 1mm, depth 0, at the same offset that's currently in configuration. Then we can pull it out of configuration entirely!

vegano1 committed:
add second well B1 to represent the z offset when getting the structure calibration height

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_heatershaker_module/1.json:
```diff
@@ -0,0 +1,62 @@
+      "diameter": 1,
+      "x": 13,
+      "y": 13,
+      "z": 0
```
This probably isn't right - this wants to be 13.85, the same as A1, since it specifies the top plane of the well

sanni-t replied:
The z here is supposed to bottom plane of the well. But I don't know what this well is so not sure what the z should be here.

sfoster1 replied:
wait it is? I've definitely been giving the opposite recommendation the entire time... I thought it was the top center

vegano1 replied:
@sfoster1 

The B1 well represents the offset we apply to the A1 well to find our probe location, don't we want to keep our z at the same height by not adding an offset?

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_magnetic_module/1.json:
```diff
@@ -0,0 +1,62 @@
+      "diameter": 1,
+      "x": 13,
+      "y": 13,
+      "z": 0
```
Probably wants to be 15.5

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_temperature_module/1.json:
```diff
@@ -0,0 +1,62 @@
+      "diameter": 1,
+      "x": 13,
+      "y": 13,
+      "z": 0
```
Probably wants to be 15

sfoster1 commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_thermocycler_module/1.json:
```diff
@@ -0,0 +1,62 @@
+      "diameter": 1,
+      "x": 13,
+      "y": 13,
+      "z": 0
```
Probably wants to be 15

sfoster1 reviewed (changes requested):
Looking good, the new wells look great. I think the Z of the new well is wrong though.

sanni-t commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_heatershaker_module/1.json:
```diff
@@ -0,0 +1,62 @@
+  },
+  "metadata": {
+    "displayName": "Opentrons Calibration Adapter - Heater-shaker Module",
+    "displayCategory": "aluminumBlock",
```
I thought we were not going to call these `aluminumBlock`?

sanni-t commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_heatershaker_module/1.json:
```diff
@@ -0,0 +1,62 @@
+      "shape": "rectangular",
+      "xDimension": 20,
+      "yDimension": 20,
+      "x": 64,
+      "y": 43,
+      "z": 13.85
```
This is supposed to be the (x,y,z) coordinates of the bottom-center point of the well. So for a well that's 4.75m mm deep in a labware of height 13.85, the z coordinate comes out to be 9.1mm.

sanni-t commented on shared-data/labware/definitions/2/opentrons_calibration_adapter_heatershaker_module/1.json:
```diff
@@ -0,0 +1,62 @@
+  "version": 1,
+  "schemaVersion": 2,
+  "cornerOffsetFromSlot": {
+    "x": 0,
```
(The below is how I think this definition should be, but I am not fully sure whether our geometry system tries to center a labware over a slot so going to find this out.)
I think this should specify an x offset since the calibration block doesn't span across the whole slot but is smaller and sits on the slot with an offset from the left/right edges. 
I believe the way this definition is written, if we were to render a drawing out of it, it would show a rectangle smaller than the slot but not centered over the slot.. it would be touching the left edge of the slot. And the well would be at the center of the slot but not at the center of the calibration block.
If we write the well as
```
"A1": {
      "depth": 4.75,
      "totalLiquidVolume": 0,
      "shape": "rectangular",
      "xDimension": 20,
      "yDimension": 20,
      "x": 55.5,
      "y": 37.5,
      "z": 9.1
      }
```
and specify a `cornerOffsetFromSlot` of (8.5, 5, 0) (Considering a slot dimension of 128)

vegano1 replied:
@sanni-t 
The two wells represent
1. A1 - The vector from the labware origin to the center of the calibration square
2. B1 - The offset to apply to the A1 well to find the z height of the calibration square

This is because the first well is the center of the calibration block, so if we were to probe that exact location we would not be able to hit the calibration block. This is why we have the second well to serve as the offset, so we can probe the edge of the calibration square instead of the center.

Is this the correct way to represent this data?

sfoster1 commented:
Yeah I've been totally wrong the entire time, it looks like! The well coordinate is to the bottom of the well.

vegano1 committed:
- fix the well position so its the distance from the labware bottom-left edge to the well center
- made the calibration point well (B1) follow the same definition of the distance from the labware bottom-left to center of the well. This will be our direct calibration point and already includes the 13x13 offset
- added cornerOffsetFromSlot since the center of the labware is not the center of the slot

vegano1 commented:
@sfoster1 
After speaking with @sanni-t I made the following changes

- Fixed the well position (A1) so it measures the distance from the labware bottom-left edge to the well center per the schema definition.
- Made the calibration point well (B1) follow the same definition of the distance from the labware bottom-left to the center of the well. This will be our direct calibration point as it already includes the 13x13mm offset.
- Added `cornerOffsetFromSlot` since the center of the calibration labware is not the center of the deck slot.

sfoster1 reviewed (approved):
Okay, nice! Thank you @sanni-t
